src/main/java/Duke.java

```diff
@@ -1,10 +1,51 @@
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Loads the tasks from Path.
+     * @param filePath Path containing the text file to be read from.
+     */
+    public Duke(Path filePath) {
+        ui = new Ui();
+        try {
+            storage = new Storage(filePath);
+            tasks = storage.load();
+
+        } catch (IOException e) {
+            System.out.println("File loading error");
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts the program.
+     */
+    public void run() {
+        tasks = ui.takeInput(tasks);
+        try {
+            storage.save(tasks);
```

Comment:
Not sure if this matters, but the project said to save after any changes to the list instead of just when the program closes